hopsworks-api/src/main/java/io/hops/hopsworks/api/experiments/dto/ExperimentDTO.java

```diff
@@ -66,7 +66,7 @@ public ExperimentDTO() {
 
   private String bestDir;
 
-  private String[] environment;
+  private String environment;
```

Comment:
If you run an experiment on a cluster with multiple machine types, for example CPU and GPU machines, it will export a .yml for each MachineType. So at most this would be of size 2 in the current implementation. However after gautiers work we don't need Machine Types anymore. So this field can be converted to a String, but for now let's keep it an array: I'm guessing @berthoug  is not removing Machine Types with this PR. No expat needed.

Reply:
Sounds good, it makes sense to change it to a String then and create JIRA to remove deprecated "machine types" code

Reply:
I'm a little confused, I understand that @robzor92 says that I should let it as an array for now and that @tkakantousis says that we should change it to a string and handle the remaining of the refactor later. So which one is it? :)

Reply:
@robzor92 wrote > but for now let's keep it an array: I'm guessing @berthoug is not removing Machine Types with this PR. No expat needed.

Which from what I understand means, If we don't remove the Machine Types in this PR we let it as an array, otherwise we change it to String.

Let's keep it an array to make things progress but open a JIRA documenting that we need to remove the Machine Types and that expat work is needed for that

Reply:
Yes leave it as an array. I will open a jira to remove Machine Types.

Reply:
Ok